pkg/requester/selection/switch.go

```diff
@@ -0,0 +1,49 @@
+package selection
+
+import (
+	"context"
+
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+	"github.com/bacalhau-project/bacalhau/pkg/requester"
+)
+
+type switcher struct {
+	selectors map[bool]requester.NodeSelector
+}
+
+// Returns a node selector that switches between node selectors dependent on the
+// targeting type defined in the job.
+func NewNodeSelectorSwitch(forAny, forAll requester.NodeSelector) requester.NodeSelector {
```

Comment:
Do love this idea, and doing it in the same vein as the executors where the `Deal` is actually just a CID to a schema that describes the deal, pluggable deals basically. And if we do the work to put WASM in the executor spec (e.g. to allow validation in the CLI of pluggable executors) then this would be another place we could use that feature.